src/main/java/org/kohsuke/github/GHPullRequest.java

```diff
@@ -629,6 +631,132 @@ public enum MergeMethod {
         REBASE
     }
 
+    /**
+     * Get pull request id for GraphQL
+     *
+     * @return The pull request id for GraphQL
+     * @throws IOException
+     *             the io exception
+     */
+    public String getGraphqlPullRequestId() throws IOException {
+        if (owner == null) {
+            throw new IllegalStateException("Repository owner is required to get the pull request ID");
+        }
+        String repositoryName = owner.getName();
+        String ownerName = owner.getOwnerName();
+
+        String graphqlBody = String.format(
+                "query GetPullRequestID { repository(name: \"%s\", owner: \"%s\") { pullRequest(number: %d) { id } } }",
```

Comment:
I think this GraphQL `id` field is already available in `GHPullRequest` from `getNodeId()` (inherited from `GHObject`).